doc: LICENSING: Add BSD-3-clause for python-devicetree test files (#86577)
zephyrproject-rtos/zephyr: doc/LICENSING.rst

  * :zephyr_file:`modules/openthread/platform/radio_spinel.cpp`
  * :zephyr_file:`modules/openthread/platform/hdlc_interface.cpp`

Python Devicetree library test files
------------------------------------

* *Licensing:* `BSD-3-clause`_

> * *Impact:* These are only used for testing and never linked in a firmware.

Reviewer: ```suggestion
* *Impact:* These are only used for testing and never linked with the firmware.
```
?
